lib/galaxy/webapps/galaxy/api/histories.py

```diff
@@ -317,9 +317,16 @@ def create(self, trans, payload, **kwd):
 
         if "archive_source" in payload:
             archive_source = payload["archive_source"]
-            archive_type = payload.get("archive_type", "url")
+            archive_file = payload.get("archive_file")
+            if archive_source:
+                archive_type = payload.get("archive_type", "url")
+            elif hasattr(archive_file, "file"):
+                archive_source = payload["archive_file"].file.name
+                archive_type = "file"
+            else:
+                raise exceptions.MessageException("Please provide a url or file.")
```

Comment:
Thanks for the backend fix - sorry about that issues!

Reply:
No worries, thx for looking into it. It was helpful. I wonder if I can further improve the handling or if its good enough that archive_source is passed empty and then filled through the name in archive_file.